### ISSUE TITLE
BUG: Does not work with Node 20 LTS

### ISSUE THREAD

**Issue by janthony6** [mkormendy/homebridge-konnected#40]
With Node 20, konnected doesn't read any contact sensor states. I can trigger the alarm manually in the HomeKit app, but nothing else works. Door open when the alarm is armed won't trigger the alarm. With node 20, a fresh install of a pro board won't pair either.

**Comment by racerx9146** [comment]
I believe I am having the same issue. I am on v20.9.0

**Comment by mkormendy** [comment]
Correct, this plugin has not been updated to work with Node 20. I will endeavor to work on this in January as I finally have a home to test the boards against the installed alarm system here.

**Comment by racerx9146** [comment]
Many thx Mike. Happy new yearOn Dec 27, 2023, at 8:42 AM, Mike Kormendy ***@***.***> wrote:﻿
Correct, this plugin has not been updated to work with Node 20. I will endeavor to work on this in January as I finally have a home to test the boards against the installed alarm system here.

**Comment by benkemal** [comment]
Hi @mkormendy, just wanted to check if any update on Node 20 support? 

I'm having issues with my setup when trying to roll back to homebridge version on Node 18, so just wanted to check if this issue is likely to be be updated soon before I spend too much more time trying to rebuild my setup.

**Comment by mkormendy** [comment]
Good news, I've recently upgraded my system that I do my development with to Node 21 and slowly upgrading all of my other applications that use v20+.

I'd like to also make this backward compatible so people who are locked in at the lowest supported version that the current code is running on, is not affected by any breaking changes introduced in the v20+ update.

I'll have to write some deployment tests to verify that too.

**Comment by darrenmadams** [comment]
I believe from reading some other forums that I am having the same problem.  I'm getting an error "The panel at http://192.168.4.80:15300/settings has disconnected and is likely rebooting to apply new provisioning settings" and after resetting everything on the hardware, reinstalling homebridge-konnected, re-adding to Home I don't have any sensors being created or triggered.

I was able to roll back cleanly by running "sudo hb-service update-node 18.18.0" but that broke other plugins.  I tried to use npm to rebuild but it just hung so I uninstalled and reinstalled and reconfigured them and it looks like they're still complaining so it would be nice to have support for v20 in this plugin.  Thanks!

**Comment by janthony6** [comment]
homebridge wants us all to be on node20. is there any way to release the node20 compatible version. anyone that has it broken by that update can easily update to node20.

**Comment by mkormendy** [comment]
So I've upgraded my dev setup to v21.7.1 and I can't seem to reproduce the problem.

**Comment by HughesAndrewC** [comment]
I just updated from 18.19.0 to 20.11.1 and the beeper-switches I have configured no longer work. I also have the same problem as above, the alarm will no longer trigger after opening a door while the alarm is armed. Downgrading back to 18.19.0 resolved both issues. Lucky for me, other plugins I'm running do not seem to depend on v20.

**Comment by mkormendy** [comment]
As asked, please update to 21.7.1 and let me know if that fixes things.

**Comment by benkemal** [comment]
@mkormendy per the [Homebridge wiki](https://github.com/homebridge/homebridge/wiki/How-To-Update-Node.js), it mentions that "Homebridge does not support odd-numbered releases of Node.js, such as 19.x and 21.x."

Any chance we can get a fix that supports v20? I don't want this to break other plugins in use and also as I'm running Homebridge on Synology, I don't think it's currently possible for me to upgrade to v21 as I don't think they have released a node v21 package.

**Comment by mkormendy** [comment]
@benkemal I've successfully reproduced the problem, and it has to do with a number of breaking changes that occur in packages that need upgrading to work with Node20. Unfortunately this has also affected my plugin severely as well, and I have gone to "the point of no-return" - meaning I am actively rewriting/refactoring the plugin to bring it up to current standards.

**Comment by mkormendy** [comment]
I believe the problem is now related to this: https://github.com/node-fetch/node-fetch/issues/1735 .. looking into fixes to node-fetch and/or using Axios.
